### PR TITLE
Add websocketDocID to error so that it gets logged in telemetry

### LINKED ISSUE #4143: socketDocumentId  on connection errors
We have a bunch of reports where we fail to connect to ordering service in SPO driver and we have no correlation ID to connect with server logs.
We have successful joinSesison call, and thus ISocketStorageDiscovery.id that we report (on successful path) as socketDocumentId in the rest of our telemetry.
But here, connection fails, so we never report any correlation ID (this one, or clientID), and thus we have nothing to work with.

Can OdspDocumentService.connectToDeltaStream() add socketDocumentId  property to error object before it re-throws it? 

I believe that's enough to get all the data we need.

If possible, we should fast-track it into 0.27 to faster understand the problem.

### PR DESCRIPTION
**Pull request by jatgarg**
Fixes: https://github.com/microsoft/FluidFramework/issues/4143

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspDocumentService.ts [hunk 1]
```diff
@@ -265,6 +265,7 @@ export class OdspDocumentService implements IDocumentService {
                 return connection;
             } catch (error) {
                 this.cache.sessionJoinCache.remove(this.joinSessionKey);
+                error.socketDocumentId = websocketEndpoint.id;
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentService.ts:268]: I'd add some safety here and at least test that it's an object and not null.

Comment (vladsud) [reply packages/drivers/odsp-driver/src/odspDocumentService.ts:268]: Also can you please check that in all core cases we actually use object derived from CustomErrorWithProps class? As otherwise properties will not propagate to telemetry

**Review by vladsud — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspDocumentService.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentService.ts:268]: hmm, I thought we always through uniform errors. I  will add check.

File: packages/drivers/odsp-driver/src/odspDocumentService.ts [hunk 2]
```diff
@@ -265,6 +265,9 @@ export class OdspDocumentService implements IDocumentService {
                 return connection;
             } catch (error) {
                 this.cache.sessionJoinCache.remove(this.joinSessionKey);
+                if (error) {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentService.ts:268]: if (typeof error === "object" && error !== null)

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentService.ts]: Yes, I was adding this only. Thanks.

**Comment by jatgarg** [comment]
Also do we need to put this in 0.27? @vladsud

File: packages/drivers/odsp-driver/src/odspDocumentService.ts (hunk 1 above)
Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentService.ts]: Yes, we are throwing created with properties.

**Comment by vladsud** [comment]
I think we have got more clarity on why things break, so no, no need to rush into 0.27
